### ISSUE TITLE
Try loss-re-weighting to reduce validation loss growth & high-frequency noise on slow variables

### ISSUE THREAD

**Issue by jder** [m2lines/Samudra#302]
https://m2lines.slack.com/files/U086XUETAJ0/F0952H4G3GF/dynamic_loss_weighting.pptx

**Comment by jder** [comment]
ok, we have some [low res proof-of-concepts](https://wandb.ai/m2lines/ocean-emulators/workspace?nw=05l9jpavwgjy) which suggest to me we should:

- [x] Would be nice to be able to see train mse [edit: did unscaled loss]
- [x] Fix restoration of loss state
- [ ] ~Make "best val checkpoint" use mse, not scaled losses~ [edit: let's not do this, we care about the scaled loss]
- [ ] ~Expose an mse per channel and overall~ [edit: reported val mses are not normalized, decided to punt]
- [x] Merge #313 and verify it works for restoring EMA state
- [x] Run a high-res version

**Comment by jder** [comment]
For posterity, here's how I've been setting up these runs on the low-res data (8x80GB A100s)

```
#!/bin/bash

cat > ~/.bash_profile <<EOF
ROOT_DIR=$(pwd)
BRANCH=u/jder/mse-dynamic-loss
CONFIG=configs/train_om4.yaml
SHARED_FS_NAME=jder-fs-1
EOF
source ~/.bash_profile

curl -LsSf https://astral.sh/uv/install.sh | sh

curl https://rclone.org/install.sh | sudo bash
rclone config create nyu-osn s3 provider=Other endpoint=https://nyu1.osn.mghpcc.org/ --all # will prompt for auth

tmux

mkdir data
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4_means data/public/OM4_means.zarr
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4_stds data/public/OM4_stds.zarr
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4 data/public/OM4.zarr --transfers=32

git clone git@github.com:Open-Athena/Ocean_Emulator.git
cd Ocean_Emulator
git switch $BRANCH

uv run torchrun --standalone --nnodes=1 --nproc-per-node=auto -m ocean_emulators.train $CONFIG --experiment.data_root=$ROOT_DIR/data/public --experiment.base_output_dir=$ROOT_DIR/$SHARED_FS_NAME --experiment.wandb.mode=online --loss=mse_dynamic --data=@configs/data/public.yaml --samudra.corrector.ocean_heat_corrector=false --experiment.name=mse_dynamic --samudra.checkpointing=simple --data.num_workers=4
```

**Comment by jder** [comment]
and here's what I did for the eval run, approximately:

```
#!/bin/bash

cat > ~/.bash_profile <<EOF
ROOT_DIR=$(pwd)
BRANCH=u/jder/mse-dynamic-loss
CONFIG=configs/eval_om4.yaml
SHARED_FS_NAME=jder-fs-1
TARGET_CHECKPOINT=$ROOT_DIR/$SHARED_FS_NAME/mse_dynamic_no_limit/saved_nets/ema_ckpt.pt
EOF
source ~/.bash_profile

curl -LsSf https://astral.sh/uv/install.sh | sh

curl https://rclone.org/install.sh | sudo bash
rclone config create nyu-osn s3 provider=Other endpoint=https://nyu1.osn.mghpcc.org/ --all # will prompt for auth

tmux

mkdir data
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4_means data/public/OM4_means.zarr
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4_stds data/public/OM4_stds.zarr
rclone copy --progress nyu-osn:m2lines-pubs/Samudra/OM4 data/public/OM4.zarr --transfers=32

git clone git@github.com:Open-Athena/Ocean_Emulator.git
cd Ocean_Emulator
git switch $BRANCH

uv run -m ocean_emulators.eval $CONFIG --experiment.data_root=$ROOT_DIR/data/public --ckpt_path=$TARGET_CHECKPOINT --experiment.wandb.mode=online --data=@configs/data/public.yaml --experiment.name=mse_dynamic_eval --num_model_steps_forward=25 --experiment.base_output_dir=$ROOT_DIR/$SHARED_FS_NAME
```

**Comment by jder** [comment]
ok, the high res version is here: https://wandb.ai/m2lines/ocean-emulators/runs/5b45eqz2?nw=nwuseroa_jder -- I configured it to use lots of epochs but stopped it after 87 epochs because it was not having the same behavior, probably because using many epoch caused the learning rate to stay high. Gonna re-run with a learning rate that comes down more quickly.

**Comment by jder** [comment]
This is now part of our baseline run. https://openathena.slack.com/archives/C08CYM42DT3/p1755882950711999
